packages/react-server/src/ReactServerStreamConfigBrowser.js

```diff
@@ -13,7 +13,7 @@ export type PrecomputedChunk = Uint8Array;
 export type Chunk = Uint8Array;
 
 export function scheduleWork(callback: () => void) {
-  callback();
+  setTimeout(callback, 0);
```

Comment:
Maybe use `setImmediate` if available? Like in Scheduler: https://github.com/facebook/react/blob/main/packages/scheduler/src/forks/Scheduler.js#L551-L580